sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,269 @@
+"""
+    Copyright Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Launch a SageMaker endpoint for realtime inference
+"""
+
+import os
+import argparse
+import logging
+import tempfile
+import boto3 # pylint: disable=import-error
+from time import gmtime, strftime
+import sagemaker as sm
+from botocore.exceptions import WaiterError
+
+from launch_utils import (wrap_model_artifacts,
+                          check_tarfile_s3_object,
+                          parse_s3_url,
+                          upload_data_to_s3,
+                          check_name_format)
+
+
+# TODO: When adding new realtime inference tasks, modify this list
+SUPPORTED_REALTIME_INFER_TASKS = ['node_classification']
+
+# Constants for SageMaker endpoints
+ENTRY_FOLDER_NAME = 'realtime_entry_points'
+ENTRY_FILE_NAMES = [
+    'node_classification_entry.py',
+    'node_regression_entry.py',
+    'edge_classification_entry.py',
+    'edge_regression_entry.py']
+DEFAULT_GS_MODLE_FILE_NAME = 'model.pt'
+
+
+def run_job(input_args):
+    """ The procedure of deploying a SageMaker real-time inference endpoint
+    
+    Following SageMaker's document for deploying model for real-time inference in
+    https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints-deploy-models.html,
+    the general steps of the deployment include:
+    1. prepare model artifacts;
+    2. create a deployable model;
+    3. create an endpoint configuration based on the deployable model;
+    4. create an endpoint based on the endpoint configuration.
+
+    This job follows these steps.
+    """
+    # ================= prepare model artifacts ================= #
+    # prepare sessions
+    b3_session = boto3.Session(region_name=input_args.region)
+    sm_session = sm.Session(boto_session=b3_session)
+    
+    # If users provide the S3 tar file path already, directly use it for model creation
+    if input_args.model_tarfile_s3:
+        entrypoint_file_name = input_args.entrypoint_file_name
```

Comment:
Do we need to check the existance of the entrypoint file?

Reply:
For custom models, it will be users' responsibility to make sure the correctness of the tar file. So we may not need to check the tar file contents.